prometheus/promhttp/http.go

```diff
@@ -169,15 +170,31 @@ func HandlerForTransactional(reg prometheus.TransactionalGatherer, opts HandlerO
 		header.Set(contentTypeHeader, string(contentType))
 
 		w := io.Writer(rsp)
-		if !opts.DisableCompression && gzipAccepted(req.Header) {
-			header.Set(contentEncodingHeader, "gzip")
-			gz := gzipPool.Get().(*gzip.Writer)
-			defer gzipPool.Put(gz)
+		if !opts.DisableCompression {
+			// Gzip takes precedence over zstd
+			// TODO(mrueg): Replace klauspost/compress with stdlib implementation once https://github.com/golang/go/issues/62513 is implemented.
+			if EncodingAccepted(req.Header, "zstd") {
```

Comment:
Something is off here. Shouldn't we take whatever is first on "Accept-Encoding" (or what has the biggest q?).

Also as it's implemented here zstd takes precedence I think ;p

Reply:
I've switched it to an actual parser that does the negotiation.